src/rockthevote.cpp

```diff
@@ -0,0 +1,494 @@
+		g_ExtendState = EExtendState::EXTEND_ALLOWED;
+	
+	g_ExtendsLeft += 1;
+
+	ClientPrintAll(HUD_PRINTTALK, CHAT_PREFIX ADMIN_PREFIX "allowed for an additional extend.", pszCommandPlayerName);
+}
+
+CON_COMMAND_CHAT(extendsleft, "Display amount of extends left for the current map")
+{
+	char message[64];
+
+	switch (g_ExtendsLeft)
+	{
+	case 0:
+		strcpy(message, "There are no extends left.");
+		break;
+	case 1:
+		strcpy(message, "There's 1 extend left");
+		break;
+	default:
+		V_snprintf(message, sizeof(message), "There are %i extends left.", g_ExtendsLeft);
+		break;
+	}
+
+	if (player)
+		ClientPrint(player, HUD_PRINTTALK, CHAT_PREFIX "%s", message);
+	else
+		ConMsg("%s", message);
+}
+
+CON_COMMAND_CHAT(timeleft, "Display time left to end of current map.")
+{
+	if (!player)
+	{
+		ClientPrint(player, HUD_PRINTCONSOLE, CHAT_PREFIX "You cannot use this command from the server console.");
+		return;
+	}
+
+	ConVar* cvar = g_pCVar->GetConVar(g_pCVar->FindConVar("mp_timelimit"));
+
+	float flTimelimit;
+	memcpy(&flTimelimit, &cvar->values, sizeof(flTimelimit));
+
+	if (flTimelimit == 0)
+	{
+		ClientPrint(player, HUD_PRINTTALK, CHAT_PREFIX "No time limit");
+		return;
+	}
+
+	int iTimeleft = (int) ((g_pGameRules->m_flGameStartTime + flTimelimit * 60.0f) - gpGlobals->curtime);
+
+	if (iTimeleft < 0)
+	{
+		ClientPrint(player, HUD_PRINTTALK, CHAT_PREFIX "Last round!");
+		return;
+	}
+
+	div_t div = std::div(iTimeleft, 60);
+	int iMinutesLeft = div.quot;
+	int iSecondsLeft = div.rem;
+	
+	if (iMinutesLeft > 0)
+		ClientPrint(player, HUD_PRINTTALK, CHAT_PREFIX "Timeleft: %i minutes %i seconds", iMinutesLeft, iSecondsLeft);
+	else
+		ClientPrint(player, HUD_PRINTTALK, CHAT_PREFIX "Timeleft: %i seconds", iSecondsLeft);
+}
+
+// Set amount of extends for the current map based on the 'maps' config file
+void SetExtendsLeft()
+{
+	KeyValues* pKV = new KeyValues("maps");
+	KeyValues::AutoDelete autoDelete(pKV);
+
+	const char* pszPath = "addons/cs2fixes/configs/maps.cfg";
+
+	const char* pszMapname = gpGlobals->mapname.ToCStr();
+
+	if (!pKV->LoadFromFile(g_pFullFileSystem, pszPath))
+	{
+		Warning("Failed to load %s, defaulting to 1 extend for current map.\n", pszPath);
+		g_ExtendsLeft = 1;
+		return;
+	}
+	for (KeyValues* pKey = pKV->GetFirstSubKey(); pKey; pKey = pKey->GetNextKey())
+	{
+		const char* pszName = pKey->GetName();
+		
+		if (strcmp(pszName, pszMapname) != 0)
+			continue;
+		
+		int iExtendsLeft = pKey->GetInt("extends", -1);
+
+		if (iExtendsLeft == -1)
+		{
+			ConMsg("Map entry %s is missing 'extends' key, defaulting to 1 extend\n", pszMapname);
+			g_ExtendsLeft = 1;
+			return;
+		}
+
+		if (iExtendsLeft == 0)
+			g_ExtendState = EExtendState::NO_EXTENDS;
+
+		ConMsg("Setting %i extend(s) for map: %s\n", iExtendsLeft, pszMapname);
+		g_ExtendsLeft = iExtendsLeft;
+		return;
+	}
+
+	ConMsg("Could not find map \"%s\" in %s, defaulting to 1 extend\n", pszMapname, pszPath);
```

Comment:
Do we really need to have a dedicated cfg just for map extends? Right now you can hardcode to 1-2 extends until we get convar support, then map configs will change the extend amount from there.